third_party/terraform/tests/resource_compute_per_instance_config_test.go

```diff
@@ -24,7 +24,7 @@ func TestAccComputePerInstanceConfig_statefulBasic(t *testing.T) {
 		"config_name4":  fmt.Sprintf("instance-%s", randString(t, 10)),
 	}
 	igmId := fmt.Sprintf("projects/%s/zones/%s/instanceGroupManagers/%s",
-		getTestProjectFromEnv(), "us-central1-c", igmName)
+		getTestProjectFromEnv(), getTestZoneFromEnv(), igmName)
```

Comment:
This would change how this test works in CI; `getTestZoneFromEnv()` would return `us-central1-a` instead of `us-central1-c`. I believe that is intentional, so I'm wary of changing it. What is the reason behind this change?

Reply:
(And the change to https://github.com/GoogleCloudPlatform/magic-modules/pull/4377/files#diff-17d67e8075e3c175aa6067907a0addfcd350422163df7569340cd1cbe61714b3L296)

Reply:
I don't see why this test should run in c. This is what caused the weird failure I mentioned right at the top. The provider loaded a from the environment but c was hardcoded and the test failed.

Reply:
I don't see a particular reason to run it in `us-central1-c` but I'm definitely concerned about having a bug where having an explicit zone specified causes strange behavior like you mentioned above; I would strongly prefer sticking with us-central1-c and figuring out the root cause of the bug. Unfortunately, I don't seem to be able to reproduce the issue locally.

Reply:
Look at this function more closely. https://github.com/GoogleCloudPlatform/magic-modules/blob/8b6e708adb80f92ce06af6ce3ac1a1ecd31e7e22/third_party/terraform/tests/resource_compute_per_instance_config_test.go#L128

In line 296 a zone(us-central1-c) is specified, but I removed the zone in line 130 and it inferred us-central1-a from the provider(set via env variable). That causes the per_instance_config to fail creation as we dont have a MIG in us-central1-a, which is why I don't like the zone being hardcoded especially in complex finegrained tests.

Reply:
Ahhh, so the issue above was that the configs for this test were inferring the provider but the test itself wasn't, so there was a mismatch. Gotcha, thanks!

@slevenick it looks like you introduced `us-central1-c` in https://github.com/GoogleCloudPlatform/magic-modules/pull/3440 - do you happen to remember if there was a specific reason to use `us-central1-c` as opposed to `us-central1-a`?

Reply:
Nope no idea. Probably doesn't matter

Reply:
cool, then let's go ahead with this change :-)